File: Evergrande/src/gestorAplicacion/otros/Pago.java
package gestorAplicacion.otros;

import java.io.Serializable;
import java.util.ArrayList;
import gestorAplicacion.herencia.Inmueble;

public class Pago implements Serializable{
    private int idPago;
    private double valor;
    private int ano;
    private int mes;
    private Inmueble inmueble;

    private static ArrayList<Pago> pagos = new ArrayList<Pago>();

    public Pago(){
        this.idPago = generateId();
        pagos.add(this);
    }

    public Pago(double valor, int ano, int mes, Inmueble inmueble){
        this.valor = valor;
        this.ano = ano;
        this.mes = mes;
        this.inmueble = inmueble;

        this.idPago = generateId();
        pagos.add(this);
    }

    private static int generateId(){
        return pagos.size() + 1;
    }

    // metodo para ver todos los pagos
    public static ArrayList<Pago> verPagos(){
        return pagos;
    }

    // metodo para filtrar los pagos por inmueble
    public static ArrayList<Pago> verPagos(Inmueble inmueble){
        ArrayList<Pago> pagosFiltro = new ArrayList<Pago>();

        for (Pago pago : pagos) {
            if (pago.inmueble.getIdInmueble() == inmueble.getIdInmueble()) {
                pagosFiltro.add(pago);
            }
        }

        return pagosFiltro;
    }

    public static void setPagos(ArrayList<Pago> pagos){
        Pago.pagos = pagos;
    }

    //  setters y getters
    
    public int getIdPago() {
        return idPago;
    }
    public void setIdPago(int idPago) {
        this.idPago = idPago;
    }
    public double getvalor() {
        return valor;
    }
    public void setvalor(double valor) {
        this.valor = valor;
    }
    public int getAno() {
        return ano;
    }
    public int getMes() {
        return mes;
    }
    public void setFecha(int ano, int mes) {
        this.ano = ano;
        this.mes = mes;
    }
    public Inmueble getInmueble() {
        return inmueble;
    }
    public void setInmueble(Inmueble inmueble) {
        this.inmueble = inmueble;
    }
}

File: Python/src/gestorAplicacion/otros/Pago.py
'''
Autores: 
Julián Orozco Vanegas
Juan Nicolas Piedrahita Salas
David Escobar Ruiz

Esta clase simula el comportamiento de un pago realizado de un cliente a un inmueble y sus caracteristicas
'''

class Pago:
    _pagos = []

    def __init__(self, valor, ano, mes, idInmueble):
        self._valor = valor
        self._ano = ano
        self._mes = mes
        self._idInmueble = idInmueble

        self._idPago = len(Pago._pagos) + 1
        Pago._pagos.append(self)
    
    @classmethod
    def verPagos(cls, idInmueble = None):
        
        if idInmueble != None:
            return list(filter(lambda pago: pago._idInmueble == idInmueble, cls._pagos))
        else:
            return cls._pagos

    @classmethod
    def setPagos(cls, pagos):
        cls._pagos = pagos

    @classmethod
    def getPagos(cls):
        return cls._pagos

    # getters y setters

    def getValor(self):
        return self._valor
    def getAno(self):
        return self._ano
    def getMes(self):
        return self._mes
    def getIdInmueble(self):
        return self._idInmueble
    def getId(self):
        return self._idPago

    def setValor(self, valor):
        self._valor = valor
    def setAno(self, ano):
        self._ano = ano
    def setMes(self, mes):
        self._mes = mes
    def setIdInmueble(self, idInmueble):
        self._idInmueble = idInmueble